### PR TITLE
binary search

### PR DESCRIPTION
**Pull request by KaroukinStanislau**
binary search first commit

### CODE REVIEW

File: search/script.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+   var a = [0, 1, 2, 2, 3, 4, 11, 22, 69, 235, 459, 637, 789, 854];
```
Comment (NikitaSedyx) [inline search/script.js]: wrong indents

File: search/script.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+   console.log(search(a, 235));
+   console.log(search(a, 0));
+   console.log(search(a, 854));
+
+   function search(array, value) {
+     return recursive(array, value, 0, array.length);
+   };
+
+   function recursive(array, value, left, right) {
```
Comment (NikitaSedyx) [inline search/script.js]: I'd suggest to move it under search function, it's helper and it shouldn't be in global scope

File: search/script.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+   a.sort(function(a, b) {
+     return a - b;
+   })
+
+   console.log(search(a, 235));
+   console.log(search(a, 0));
+   console.log(search(a, 854));
+
+   function search(array, value) {
+     return recursive(array, value, 0, array.length);
```
Comment (NikitaSedyx) [inline search/script.js]: btw is it possible for you to implement function without helper?

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

File: search/script.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+var a = [0, 1, 2, 2, 3, 4, 11, 22, 69, 235, 459, 637, 789, 854];
+
+console.log(recursive(a, 235));
+console.log(recursive(a, 0));
+console.log(recursive(a, 854));
```
Comment (NikitaSedyx) [inline search/script.js]: could you try to call the same but when `a = [0, 1, 2, 2, 3, 4, 11, 22, 69, 235, 459, 637];`
looks like it returns index of last element

or with `a = [];`
all calls returns 0

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

File: search/script.js (hunk 4 above)
Comment (KaroukinStanislau) [reply search/script.js]: yes, i forgot to fix that

File: search/script.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+var a = [0, 1, 2, 2, 3, 4, 11, 22, 69, 235, 459, 637];
+
+console.log(recursive(a, 235));
+console.log(recursive(a, 0));
+console.log(recursive(a, 854));
+console.log(recursive(a, -854));
+console.log(recursive([], 0));
+
+function recursive(array, value, left, right) {
```
Comment (NikitaSedyx) [inline search/script.js]: please give another function name smth like `binarySearch`

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**